frontend/src/employee-frontend/components/unit/unit-reservations/UnitAttendanceReservationsView.tsx

```diff
@@ -78,19 +79,62 @@ export default React.memo(function UnitAttendanceReservationsView({
   groups
 }: Props) {
   const { i18n } = useTranslation()
-  const dateRange = useMemo(
-    () => getWeekDateRange(selectedDate),
-    [selectedDate]
-  )
+
+  // Before changing the week, the current week's data should be saved
+  // because it is possible the user has started adding an overnight
+  // entry over the week boundary, so the partial data should be saved
+  // before navigating to the next week. The callbacks to save the data
+  // are stored here, and added in the row components.
+  const weekSavingFns = useRef<Map<string, () => Promise<void>>>(new Map())
+
+  const [week, setWeek] = useState({
+    dateRange: getWeekDateRange(selectedDate),
+    saved: true,
+    savingPromise: Promise.resolve()
+  })
+
+  useEffect(() => {
+    let cancelled = false
+
+    const dateRange = getWeekDateRange(selectedDate)
+    setWeek({
+      dateRange,
+      saved: false,
+      savingPromise: Promise.all(
+        Array.from(weekSavingFns.current.values()).map((fn) => fn())
+      ).then(() => {
+        if (!cancelled) {
+          setWeek((week) =>
+            week.dateRange === dateRange
```

Comment:
I think this is a bit dangerous, because date ranges don't really support the equals operator and we're simply checking if it's literally the same object instance...if we *really* intend to check it like this (vs. equality based on the date values), I recommend adding a comment to explicitly communicate this is intentional

Reply:
I've added a clarifying comment; this was indeed intended, as we want to ensure the user hasn't changed the week between the saving request and response. Even if the dates values are equal, we want to check that the date range instance is the same one, so if the week is changed A -> B -> A the response should be discarded.